src/zenml/models/filter_models.py

```diff
@@ -0,0 +1,697 @@
+                getattr(table, self.column), sqlalchemy.String
+            ).startswith(f"%{self.value}%")
+        elif self.operation == GenericFilterOps.CONTAINS:
+            return cast_if(
+                getattr(table, self.column), sqlalchemy.String
+            ).endswith(f"%{self.value}%")
+        else:
+            return getattr(table, self.column) == self.value
+
+
+class NumericFilter(Filter):
+    """Filter for all numeric fields."""
+
+    value: float
+
+    ALLOWED_OPS: ClassVar[List[str]] = [
+        GenericFilterOps.EQUALS,
+        GenericFilterOps.GT,
+        GenericFilterOps.GTE,
+        GenericFilterOps.LT,
+        GenericFilterOps.LTE,
+    ]
+
+    def generate_query_conditions(
+        self,
+        table: Type[SQLModel],
+    ) -> Union["BinaryExpression", "BooleanClauseList"]:
+        """Generate the query conditions for the database.
+
+        Args:
+            table: The SQLModel table to use for the query creation
+
+        Returns:
+            A list of conditions that will be combined using the `and` operation
+        """
+        if self.operation == GenericFilterOps.GTE:
+            return getattr(table, self.column) >= self.value
+        elif self.operation == GenericFilterOps.GT:
+            return getattr(table, self.column) > self.value
+        elif self.operation == GenericFilterOps.LTE:
+            return getattr(table, self.column) <= self.value
+        elif self.operation == GenericFilterOps.LT:
+            return getattr(table, self.column) < self.value
+        else:
+            return getattr(table, self.column) == self.value
+
+
+class DatetimeFilter(Filter):
+    """Filter for all Boolean fields."""
+
+    value: datetime
+
+    ALLOWED_OPS: ClassVar[List[str]] = [
+        GenericFilterOps.EQUALS,
+        GenericFilterOps.GT,
+        GenericFilterOps.GTE,
+        GenericFilterOps.LT,
+        GenericFilterOps.LTE,
+    ]
+
+    def generate_query_conditions(
+        self,
+        table: Type[SQLModel],
+    ) -> Union["BinaryExpression", "BooleanClauseList"]:
+        """Generate the query conditions for the database.
+
+        Args:
+            table: The SQLModel table to use for the query creation
+
+        Returns:
+            A list of conditions that will be combined using the `and` operation
+        """
+        if self.operation == GenericFilterOps.GTE:
+            return getattr(table, self.column) >= self.value
+        elif self.operation == GenericFilterOps.GT:
+            return getattr(table, self.column) > self.value
+        elif self.operation == GenericFilterOps.LTE:
+            return getattr(table, self.column) <= self.value
+        elif self.operation == GenericFilterOps.LT:
+            return getattr(table, self.column) < self.value
+        else:
+            return getattr(table, self.column) == self.value
+
+
+# ---------------- #
+# PAGINATION PARAM #
+# -----------------#
+
+
+class FilterBaseModel(BaseModel):
```

Comment:
Maybe `BaseFilterModel` to keep it in line with our `BasePipeline` etc?

Reply:
Makes sense, renamed it 👍